src/components/flyout/Flyout.js

```diff
@@ -249,10 +249,18 @@ class Flyout extends React.Component<Props, State> {
             this.openOverlay();
         }
 
-        // If button was clicked, the detail field should hold number of clicks.
-        // If number is zero, the event was synthesized.
-        // https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/detail
-        const isButtonClicked = event.detail > 0;
```

Comment:
"event.detail" is the number of clicks for a mouse event. But pressing enter on a button also triggers the click event handler, so we shouldn't assume that a click count of 0 is a programmatic event. Instead, use .isTrusted.